nodes/list_mutators/vd_attr_node_mk2.py

```diff
@@ -0,0 +1,242 @@
+
+    def sv_init(self, context):
+        self.vd_init_sockets(context)
+        self.vd_init_uilayout_data(context)
+
+    def draw_buttons(self, context, layout):
+        layout.prop(context.space_data, "show_region_ui", toggle=True)
+
+    def draw_buttons_ext(self, context, layout):
+        self.draw_group(context, layout)
+
+    def ensure_correct_origin_node_name(self):
+        if self.vd_items_group:
+            if not self.vd_items_group[0].origin_node_name == self.name:
+                for item in self.vd_items_group:
+                    item.origin_node_name = self.name
+
+    def get_repr_and_socket_from_attr_name(self, attr_name):
+        socket_repr = maximum_spec_vd_dict[attr_name]
+        return socket_repr, self.inputs[socket_repr.name]
+
+    def get_attr_from_input(self, item):
+
+        socket_repr, associated_socket = self.get_repr_and_socket_from_attr_name(item.attr_name)
+
+        if associated_socket.is_linked:
+            data = associated_socket.sv_get()
+
+            if socket_repr.kind in {'4f', '3f', 'i'}:
+                return data[0][0]
+            elif socket_repr.kind in {'b'}:
+                return bool(data[0][0])
+            elif socket_repr.kind in {'enum'}:
+                prop_signature = self.vd_items_props.__annotations__[item.attr_name][1]
+                enum_index = data[0][0]
+                enum_item = prop_signature['items'][enum_index]
+                return enum_item[0]
+        else:
+            return None
+
+    def make_value_serializable(self, attr, value):
+        if maximum_spec_vd_dict[attr].kind in {'3f', '4f'}:
+            value = value[:]
+        return value
+
+    @property
+    def attrdict_from_state(self):
+
+        current_attr_dict = {}
+        if self.vd_items_group and self.vd_items_props:
+            for item in self.vd_items_group:
+                attr = item.attr_name
+                if not item.show_socket and not item.use_default:
+                    # apparantly no desire to pass this attr
+                    continue
+                if item.use_default or not item.show_socket:
+                    value = getattr(self.vd_items_props[0], attr)
+                    value = self.make_value_serializable(attr, value)
+                else:
+                    value = self.get_attr_from_input(item)
+                    if value is None:
+                        continue
+                current_attr_dict[attr] = value
+
+        return current_attr_dict
+
+    def migrate_from(self, old_node):
+        self.location.y += 150
+        raise Exception('Migration has to be done manually')
+
+    def process(self):
+        self.ensure_correct_origin_node_name()
+        self.outputs['attrs'].sv_set([self.attrdict_from_state])
+
+    ## ---- UI JSON STORAGE SECTION BELOW THIS LINE
+
+    def load_from_json(self, node_data: dict, import_version: float):
+        """ this gets triggered by IOJSON to populate this node from json """
```

Comment:
Don't create this method.
https://github.com/nortikin/sverchok/pull/3560#issuecomment-701886813

Reply:
It was already there. You want me to delete it?

Reply:
It was there for supporting old JSONs with this node because they was saved in old format. For new node this method will be useless.